dev_docs/key_concepts/persistable_state.mdx

```diff
@@ -0,0 +1,70 @@
+---
+id: kibDevDocsPersistableStateIntro
+slug: /kibana-dev-docs/persistable-state-intro
+title: Persistable State
+summary: Persitable state is a key concept to understand when building a Kibana plugin.
+date: 2021-02-02
+tags: ['kibana','dev', 'contributor', 'api docs']
+---
+
+"Persistable State" is developer defined part of state or configuration that can be(is ment to be) persisted by other
+ plugin that the one providing it. Persistable State needs to be serializable and owner can/should provide utilities
+ to migrate it, extract and inject references from/to it as well as collect telemetry.
+ 
+## exposing state that can be persisted
+
+Any plugin that exposes state or configuration that another plugin might persist should implement `PersistableStateService`
+interface on their `setup` contract. This will allow plugins persisting the state to easily access migrations and other utilities.
```

Comment:
1. if lens allows using lens without embeddable, then it should. To me it would make more sense that lens would only be exposed thru embeddable.

2. consumer: i think we should mention it in the docs, for example documentation for embeddbles should make it clear that embeddable input is persistable state. 

3. we were discussing joining migrate+inject into deserialize and extract into serialize before, but i think one of the problems was that this will not always be the same: on the server we'll need to make extraction and migration separate (due to saved object part) and on the client we won't be extracting references at all (for urls). There is also this thing about naming, the methods would not serialize/deserialize so most likely they should not be called so to avoid any confusion

Reply:
> 1.

Checked the code and it looks like Lens doesn't expose access to it directly from its start contract, so stgm. 

> 2. 

I think it's okay to say that producers should leverage comments to make the connection between persisted state and persistable state service clear, but I don't think that's a good long term solution.

> 3.

For server, in that case, we could still expose `getMigrationFunctions` and `inject` separately. Could we expose `deserialize` and `serialize` on both client and server though as a convenience function? Or perhaps just on client? Having `serialize` as part of the PersistableStateService means we can rely on typescript, not code comments, to help consumers understand the connection.